### ISSUE TITLE
upload only changes

### ISSUE THREAD

**Issue by sevenleo** [liximomo/vscode-sftp#839]
My system has a lot of files (8,967).
It is hosted on ftp and not on git.
I usually use the UploadOnSave function to synchronize changes with ftp

However I would like to test some changes locally and did not want them to go live without all the tests having been done.

So, I turn off UploadOnSave I make changes and tests then I upload everything that has been changed.

But I need to remember the changed files or upload all the files, which takes a long time.

I suggest uploading only changed files. For that, sftp would need to keep track of files changed since the last upload / sync
It's possible ? It already exists and I haven't found it?

**Comment by marcusdiy** [comment]
Yeah, sometimes changing even one file, triggers an full upload of the folder...
I have an media folder, adding just an image requires the re-uploading of the entire folder
Consuming bandwidth and time 🤦 This while using the watcher obviusly
`{ "watcher": { "files": "**/*", "autoUpload": true, "autoDelete": true } }`
